### PR TITLE
feat(functions): create new samples for typed function signature

### PR DESCRIPTION
**Pull request by garethgeorge**
Creates new samples for GCF python strongly typed function signature. Samples added are

 * greeting - a typed hello world (accepting JSON)

### CODE REVIEW

**Comment by garethgeorge** [comment]
CC @bshaffer for familiarity with the typed functions project (if you have review bandwidth)

File: functions/typed_greeting/test/TestCasesTrait.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        return [
+            [
+                'label' => 'Default',
+                'first_name' => 'Jane',
+                'last_name' => 'Doe',
+                'expected_message' => 'Hello Jane Doe!',
+            ],
+        ];
+    }
+}
```
Comment (bshaffer) [inline functions/typed_greeting/test/TestCasesTrait.php:36]: Just for simplicity I'd just put this function in the `UnitTest.php` file for simplification

**Review by bshaffer — CHANGES_REQUESTED** [review]
One suggestion - can simplify tests into into one file by copying the method over. Traits are typically for code reuse, so if you're only including the trait in one test, it doesn't really make sense. 

If you think you'll be a `SystemTest` or `DeployTest` soon (like we have in the other function samples), then we can keep them!

File: functions/typed_greeting/test/TestCasesTrait.php (hunk 1 above)
Comment (garethgeorge) [reply functions/typed_greeting/test/TestCasesTrait.php:36]: Thanks @bshaffer! Makes sense, I don't think we'll be needing more tests out of this sample.

**Review by grayside — APPROVED** [review]

File: functions/typed_greeting/test/UnitTest.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+declare(strict_types=1);
+
+namespace Google\Cloud\Samples\Functions\TypedGreeting\Test;
+
+use PHPUnit\Framework\TestCase;
+use GreetingRequest;
+use GreetingResponse;
+
+require_once __DIR__ . '/TestCasesTrait.php';
```
Comment (bshaffer) [inline functions/typed_greeting/test/UnitTest.php:26]:
```suggestion
```

**Review by bshaffer — APPROVED** [review]

**Pull request merged**